popup.js

```diff
@@ -1,34 +1,77 @@
-let text_input = document.getElementById("text_input");
-let definition = document.getElementById("definition");
-
-var background = chrome.extension.getBackgroundPage();
-let word1 = background.word;
-text_input.innerHTML = word1;
-
-const xhr = new XMLHttpRequest();
-function runWithWindow(){
-    const data = JSON.stringify([
-        {
-            "Text": word1.toString()
-        }
-    ]);
-    
-    xhr.withCredentials = true;
-    
-    xhr.open("POST", "https://microsoft-translator-text.p.rapidapi.com/translate?api-version=3.0&to=vi&textType=plain&profanityAction=NoAction");
-    xhr.setRequestHeader("content-type", "application/json");
-    xhr.setRequestHeader("x-rapidapi-key", "e84d943d16msh145795616c213d4p1071ecjsn3d924959b63c");
-    xhr.setRequestHeader("x-rapidapi-host", "microsoft-translator-text.p.rapidapi.com");
-    
-    xhr.send(data);
-};
-xhr.addEventListener("readystatechange", function () {
+document.addEventListener('DOMContentLoaded', function () {
+  const requestGetDefinition = new XMLHttpRequest();
+  const requestGetPronunciation = new XMLHttpRequest();
+
+  const audio = document.getElementById("audio");
+  const synonym = document.getElementById("synonym");
+  const textInput = document.getElementById("txt-input");
+  const definition = document.getElementById("definition");
+  const definitionEn = document.getElementById("definition-en");
+  const pronunciation = document.getElementById("pronunciation");
+
+  const background = chrome.extension.getBackgroundPage();
+
+  ////for oxford api
+  const appId = "817d8cc8";
+  const appKey = "af2a40e721acd4c13b2827a6919614f6";
+
+  const endpoint = "entries";
+  const languageCode = "en-gb";
+  const keyWord = word.toLowerCase();
+
+  const url = `https://od-api.oxforddictionaries.com/api/v2/${endpoint}/${languageCode}/${keyWord}`;
+
+  const { word } = background;
```

Comment:
const  word  = background.word;

Reply:
No. Please, use destructuring for this. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
Oh sorry I'm wrong, you're right.